bugbot/crash/analyzer.py

```diff
@@ -0,0 +1,710 @@
+        days_without_crashes: int = 7,
+    ) -> "SignaturesDataFetcher":
+        """Find new actionable crashes.
+
+        Args:
+            product: The product to check.
+            channel: The release channel to check.
+            days_to_check: The number of days to check for crashes.
+            days_without_crashes: The number of days without crashes before the
+                `days_to_check` to consider the signature new.
+
+        Returns:
+            A list of actionable signatures.
+        """
+        duration = days_to_check + days_without_crashes
+        end_date = lmdutils.get_date_ymd("today")
+        start_date = end_date - timedelta(duration)
+        earliest_allowed_date = lmdutils.get_date_str(
+            end_date - timedelta(days_to_check)
+        )
+        date_range = socorro.SuperSearch.get_search_date(start_date, end_date)
+
+        params = {
+            "product": product,
+            "release_channel": channel,
+            "date": date_range,
+            # TODO(investigate): should we do a local filter instead of the
+            # following (should we exclude the signature if one of the crashes
+            # is a shutdown hang?):
+            # If the `ipc_shutdown_state` or `shutdown_progress` field are
+            # non-empty then it's a shutdown hang.
+            "ipc_shutdown_state": "__null__",
+            "shutdown_progress": "__null__",
+            # TODO(investigate): should we use the following instead of the
+            # local filter.
+            # "oom_allocation_size": "!__null__",
+            "_aggs.signature": [
+                "moz_crash_reason",
+                "reason",
+                "_histogram.date",
+                "_cardinality.install_time",
+                "_cardinality.oom_allocation_size",
+            ],
+            "_results_number": 0,
+            "_facets_size": 10000,
+        }
+
+        def handler(search_resp: dict, data: list):
+            logger.debug(
+                "Total of %d signatures received from Socorro",
+                len(search_resp["facets"]["signature"]),
+            )
+
+            for crash in search_resp["facets"]["signature"]:
+                signature = crash["term"]
+                if any(
+                    signature.startswith(excluded_prefix)
+                    for excluded_prefix in cls.EXCLUDED_SIGNATURE_PREFIXES
+                ):
+                    # Ignore signatures that start with any of the excluded prefixes.
+                    continue
+
+                facets = crash["facets"]
+                installations = facets["cardinality_install_time"]["value"]
+                if installations <= 1:
+                    # Ignore crashes that only happen on one installation.
+                    continue
+
+                first_date = facets["histogram_date"][0]["term"]
+                if first_date < earliest_allowed_date:
+                    # The crash is not new, skip it.
```

Comment:
Should we also have a build ID check to see if the crash is new?

Reply:
Crash-Clouseau already does that. We indirectly filter out signatures from old builds since we exclude signatures not appearing on Clouseau. 

The downside of filtering signatures appearing on old builds is that it could hide bugs related to changes in the web itself, not Firefox (see https://github.com/mozilla/bugbot/issues/2122).